griffon/services/core_queries.py

```diff
@@ -285,35 +294,39 @@ def execute(self, status=None) -> List[Dict[str, Any]]:
                 params["namespace"] = self.ns
             status.update("griffoning: searching latest component(s).")
             latest_components_cnt = self.corgi_session.components.count(**params)
+            status.update(f"griffoning: found {latest_components_cnt} latest component(s).")
             latest_components = self.corgi_session.components.retrieve_list_iterator_async(
                 max_results=max_limit, **params
             )
-            status.update(f"griffoning: found {latest_components_cnt} latest component(s).")
-            for c in latest_components:
-                status.update(
-                    f"griffoning: found {latest_components_cnt} latest component(s), retrieving sources & upstreams."  # noqa
-                )
-                c.sources = async_retrieve_sources(self.corgi_session, c.purl)
-                c.upstreams = async_retrieve_upstreams(self.corgi_session, c.purl)
-                results.append(c)
+
+            with multiprocessing.Pool() as pool:
+                for processed_component in pool.map(
+                    partial(process_component, self.corgi_session), latest_components
+                ):
+                    status.update(
```

Comment:
This griffon status update should happen only once, no need to do it for every component. Multiple occurrences below.

Reply:
yes, at one point was thinking of updating that status with each component ... no longer think its worth it, will push up